src/fieldset.c

```diff
@@ -102,6 +103,64 @@ static void fs_modify_word(fieldset_t *fs, const char *name, int type,
 	fs_add_word(fs, name, type, free_, len, value);
 }
 
+static char *sanitize_utf8(const char *buf)
+{
+	const char *ptr = buf;
+
+	// Count how many errors we encounter
+	uint32_t i = 0;
+	// Upper bounds to ensure termination even if u8_check is unsafe
+	while (i < strlen(buf) && ptr < buf + strlen(buf)) {
```

Comment:
Your reaction is exactly why I wrote it as such. In really it really wants to be a while(ptr!=null), but with a maximum iteration safety belt in case u8_check behaves out of spec. I can rewrite it like a for if you'd like, but I think it makes more sense semantically as it is. Want me to change it?